# Hypothesis Testing Assignment

### 1) A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [2]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
import statsmodels.api as sm

In [10]:
cutlets = pd.read_csv("C:\\Users\\User\\Documents\\Datascience_Assignments\\Assignment 3\\Cutlets.csv")
cutlets

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

Assume Null hypothesis as Ho: μ1 = μ2

There is no difference in diameters of cutlets between two units.

Assume Alternate hypothesis as Ha: μ1 ≠ μ2 

There is significant difference in diameters of cutlets between two units

In [13]:
unitA = pd.Series(cutlets.iloc[:,0])
unitB = pd.Series(cutlets.iloc[:,1])

In [18]:
p_value = sm.stats.ttest_ind(unitA,unitB)
p_value[1]

0.47223947245995745

Given, α = 0.05.

Clearly (p_value=0.4722) > (α = 0.05). 
Accept Null Hypothesis, μ1 = μ2

### 2)    A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [19]:
tat = pd.read_csv("C:\\Users\\User\\Documents\\Datascience_Assignments\\Assignment 3\\LabTAT.csv")
tat

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [20]:
# We have nothing to compare against each other. 
# We use Anova ftest statistics for analysis of varaince containing more than 2 samples.

Assume Null Hypothesis Ho as No Variance: All samples TAT population means are same.

Assume Alternate Hypothesis Ha as It has Variance: Atleast one sample TAT population mean is different.

In [22]:
lab1 = tat.iloc[:,0]
lab2 = tat.iloc[:,1]
lab3 = tat.iloc[:,2]
lab4 = tat.iloc[:,3]

In [23]:
p_value = stats.f_oneway(lab1,lab2,lab3,lab4)
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Given, α = 0.05.

Clearly (p_value ~ 0) < (α = 0.05).

Reject Null Hypothesis or Accept Alternate Hypothesis. 

### 3) Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

![](hypo_3.png)

In [25]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

Assume Null Hypothesis as Ho: Male-Female buyer rations are similar across regions.

Assume Alternate Hypothesis as Ha: Male-Female buyer rations are dissimilar across regions.

In [28]:
buyer = np.array([[50,142,131,70],[435,1523,1356,750]])
buyer

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [32]:
result = chi2_contingency(buyer)
result[1] 

0.6603094907091882

Clearly (p_value = 0.6603) > (α = 0.05).

Accept the Null Hypothesis.

### 4) TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [33]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

Assume Null Hypothesis as Ho: Customer order forms defective % does not varies by centre. 

Assume Alternative hypothesis as Ha: Customer order forms defective % varies by centre.

In [35]:
customer = pd.read_csv("C:\\Users\\User\\Documents\\Datascience_Assignments\\Assignment 3\\Costomer+OrderForm.csv")
customer

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [36]:
customer.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [37]:
customer.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [38]:
customer.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [39]:
customer.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [40]:
order = np.array([[271,267,269,280],[29,33,31,20]])
order

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [42]:
result = chi2_contingency(order)
result[1] 

0.2771020991233135

Clearly (p_value = 0.2771) > (α = 0.05). 

Accept Null Hypothesis.